kymatio/scattering2d/backend/numpy_backend.py

```diff
@@ -22,23 +22,16 @@ def __init__(self, pad_size, input_size, pre_pad=False):
                 size of padding to apply.
             input_size : list of 2 integers
                 size of the original signal
-            pre_pad : boolean
-                if set to true, then there is no padding, one simply converts
-                from real to complex.
 
         """
-        self.pre_pad = pre_pad
         self.pad_size = pad_size
 
     def __call__(self, x):
-        if self.pre_pad:
-            return x
-        else:
-            paddings = (0, 0),
-            paddings += ((self.pad_size[0], self.pad_size[1]), (self.pad_size[2], self.pad_size[3]))
+        paddings = (0, 0),
```

Comment:
Am I missing something? Didn't this need to follow the number of axes before the last two? I.e. `paddings = (0, 0), * (x.ndim - 2)`

Reply:
We could do it that way, but we don't require this function to be batch agnostic, do we? 

Do you want it to be batch agnostic?

Reply:
1. I somehow thought in 2D we typically work with 4 axes. This here implies we work with 3
2. I don't see a problem with having some flexibility where it doesn't cost much to add and leads to readable code